src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -60,56 +60,73 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
         public static Document? GetDocument(this Solution solution, TextDocumentIdentifier documentIdentifier)
         {
             var documents = solution.GetDocuments(documentIdentifier.Uri);
-            if (documents.Length == 0)
-            {
-                return null;
-            }
-
-            return documents.FindDocumentInProjectContext(documentIdentifier);
+            return documents.Length == 0
+                ? null
+                : documents.FindDocumentInProjectContext(documentIdentifier, (sln, id) => sln.GetRequiredDocument(id));
         }
 
-        public static Document FindDocumentInProjectContext(this ImmutableArray<Document> documents, TextDocumentIdentifier documentIdentifier)
+        private static T FindItemInProjectContext<T>(
```

Comment:
merged this code to share between looking up URIs with context from docs, additional docs, and projects.